<?xml version="1.0"?>
<launch>

  <arg name="camera_info_yaml_file" default="$(find udpvid)/config/camera_info.yaml"/>
  <arg name="udpport" default="5601"/>

  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltags2_ros_continuous_node" />
  <arg name="camera_name" default="/ROVcam" />
  <arg name="camera_frame" default="ROV" />
  <arg name="image_topic" default="image_raw" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltags2_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltags2_ros)/config/tags.yaml" ns="$(arg node_namespace)" />
  

  <node pkg="udpvid" type="vidpublisher.py" name="vidpublisher" output="screen" launch-prefix="xterm -e">
	<!--node pkg="udpvid" type="vidpublisher.py" name="vidpublisher" output="screen"-->
    <param name="camera_info_yaml" value="$(arg camera_info_yaml_file)"/>
    <param name="portnum" value="$(arg udpport)"/>
  </node>

  <node pkg="apriltags2_ros" type="apriltags2_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
  
 <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view"/>

 <node pkg="rqt_graph" type="rqt_graph" name="rqt_graph"/>

</launch>
